### PR TITLE
Use conventional attribute names in the CRD

### PR DESCRIPTION
**Pull request by sbose78**
- Rename `spec.backingSelector` to `spec.BackingServiceSelector`
- Rename `spec.backingServiceSelector.resourceName` to `spec.backingServiceSelector.resourceKind`
- Rename `spec.backingServiceSelector.objectName` to  `spec.backingServiceSelector.resourceRef` 

```
apiVersion: apps.openshift.io/v1alpha1
kind: ServiceBindingRequest
metadata:
  name: example-servicebindingrequest
spec:
  backingServiceSelector:
    resourceKind: database.example.org
    resourceVersion: v1alpha1
    resourceRef: demo-mysqldb
  applicationSelector:
    matchLabels:
      connects-to: postgres
      environment: production
    resourceKind: Deployment

```

### CODE REVIEW

**Comment by ldimaggi** [comment]
Did you intend to specify an uppercase "B" here:

Rename spec.backingSelector to spec.BackingServiceSelector

File: pkg/apis/apps/v1alpha1/zz_generated.openapi.go [hunk 1]
```diff
@@ -54,13 +54,13 @@ func schema_pkg_apis_apps_v1alpha1_ApplicationSelector(ref common.ReferenceCallb
...
-func schema_pkg_apis_apps_v1alpha1_BackingSelector(ref common.ReferenceCallback) common.OpenAPIDefinition {
+func schema_pkg_apis_apps_v1alpha1_BackingServiceSelector(ref common.ReferenceCallback) common.OpenAPIDefinition {
 	return common.OpenAPIDefinition{
 		Schema: spec.Schema{
 			SchemaProps: spec.SchemaProps{
-				Description: "BackingSelector defines the selector based on resource name, version, and resource kind",
+				Description: "BackingServiceSelector defines the selector based on resource name, version, and resource kind",
 				Properties: map[string]spec.Schema{
-					"resourceName": {
+					"resourceType": {
```
Comment (Avni-Sharma) [inline pkg/apis/apps/v1alpha1/zz_generated.openapi.go]: resourceType or resourceKind ?

**Comment by Avni-Sharma** [comment]
refer #49

**Pull request closed without merging**